pkg/agent/clean/cluster.go

```diff
@@ -349,6 +355,29 @@ func cleanupRoles(client *kubernetes.Clientset) []error {
 	return errs
 }
 
+func cleanupWebhookResources(client *kubernetes.Clientset) []error {
+	logrus.Info("Starting cleanup of webhook-specific resources")
+	var errs []error
+
+	if !dryRun {
```

Comment:
This doesn't follow the pattern of the other function's use of dryRun. Notably, your implementation won't print out the resources from the webhook that it is planning on deleting, which kind of makes the dryRun a bit deceptive.

I'd refer to the below `deleteJob` method for what I mean - you can see that even in a dry run it logs out each job that it intends on removing.

Reply:
Newest changes technically achieves what I mean.